redis/redisLimiter.go

```diff
@@ -195,6 +196,57 @@ func (rl *rateLimiter) getFailConfig(mode Mode) (time.Duration, int64) {
 	return rl.freezeFailureConfig.limitPeriod, rl.freezeFailureConfig.maxFailures
 }
 
+// SetSecurityModeNoExpire will set the key from volatile to persistent
+func (rl *rateLimiter) SetSecurityModeNoExpire(key string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), rl.operationTimeout)
+	defer cancel()
+
+	rl.mutStorer.Lock()
+	defer rl.mutStorer.Unlock()
+
+	wasSet, err := rl.storer.SetPersist(ctx, key)
+
+	if err != nil {
+		return err
+	}
+
+	if !wasSet {
+		log.Debug("security mode was not set")
+	}
+
+	return nil
+}
+
+// UnsetSecurityModeNoExpire will set the key from persistent to volatile
+func (rl *rateLimiter) UnsetSecurityModeNoExpire(key string, mode Mode) error {
+	ctx, cancel := context.WithTimeout(context.Background(), rl.operationTimeout)
+	defer cancel()
+
+	rl.mutStorer.Lock()
+	defer rl.mutStorer.Unlock()
+
+	limitPeriod, _ := rl.getFailConfig(mode)
+
+	_, err := rl.storer.ExpireTime(ctx, key)
+
+	if err == nil {
+		// the key was set with a ttl by default because of a possible attack
+		_, err = rl.storer.SetExpire(ctx, key, limitPeriod)
+		return err
+	}
+
+	if !errors.Is(err, ErrNoExpirationTimeForKey) {
+		// the key doesn't exist or another error
+		log.Error("UnsetPersist", "key", key, "err", err.Error())
+		return err
+	}
+
+	// the key exists, and it was set as persist by user
+	_, err = rl.storer.SetExpire(ctx, key, limitPeriod)
+	return err
+
+}
```

Comment:
when there is no expiration time for key, it should be set to a min value or to the limit period of the Security mode?

Reply:
When unsetting the persistent security mode, I guess we want to set it with a min value, so the security mode ends completely.

In case on an ongoing attack, security mode should be automatically reached back.

@AdoAdoAdo @ssd04 what do you think?